src/model/resources/storage/long_duration_storage.jl

```diff
@@ -86,14 +96,23 @@ function long_duration_storage!(EP::Model, inputs::Dict)
 	# Build up inventory can be positive or negative
 	@variable(EP, vdSOC[y in STOR_LONG_DURATION, w=1:REP_PERIOD])
 
+			
+	# State of charge held in reserve for storage at beginning of each modeled period n
+	@variable(EP, vCAPRES_socw[y in STOR_LONG_DURATION, n in MODELED_PERIODS_INDEX] >= 0)
+
+	# Build up in storage inventory held in reserve over each representative period w
+	# Build up inventory can be positive or negative
+	@variable(EP, vCAPRES_dsoc[y in STOR_LONG_DURATION, w=1:REP_PERIOD])
+
 	### Constraints ###
 
 	# Links last time step with first time step, ensuring position in hour 1 is within eligible change from final hour position
 	# Modified initial state of storage for long-duration storage - initialize wth value carried over from last period
 	# Alternative to cSoCBalStart constraint which is included when not modeling operations wrapping and long duration storage
 	# Note: tw_min = hours_per_subperiod*(w-1)+1; tw_max = hours_per_subperiod*w
 	@constraint(EP, cSoCBalLongDurationStorageStart[w=1:REP_PERIOD, y in STOR_LONG_DURATION],
-				    EP[:vS][y,hours_per_subperiod*(w-1)+1] == (1-dfGen[y,:Self_Disch])*(EP[:vS][y,hours_per_subperiod*w]-vdSOC[y,w])-(1/dfGen[y,:Eff_Down]*EP[:vP][y,hours_per_subperiod*(w-1)+1])+(dfGen[y,:Eff_Up]*EP[:vCHARGE][y,hours_per_subperiod*(w-1)+1]))
+				    EP[:vS][y,hours_per_subperiod*(w-1)+1] == (1-dfGen[y,:Self_Disch])*(EP[:vS][y,hours_per_subperiod*w]-vdSOC[y,w])
```

Comment:
`hours_per_subperiod*(w-1)+1` and `hours_per_subperiod*w` are two expressions that appear frequently in these files. This PR does not originate them, but it would be great to add some more meaning. Could we call the former "first hour of this subperiod" and the latter "last hour of this subperiod" ? Could we write a little function like `firsthourofsubperiod(w, hours_per_subperiod) = hours_per_subperiod*(w-1)+1`  ? Is there anything that can be replaced with `hoursbefore` or `hoursafter`?

Reply:
Can we make this an issue an address in a separate PR? So that this PR only adds this one feature.

Reply:
Of course